### PR TITLE
feat: Client + contracts: disallow user to create offer if amount of resource is 0

### LINKED ISSUE #740: [client] don't allow to create order if one of the amounts is 0
image

### PR DESCRIPTION
**Pull request by edisontim**
### **User description**
Closes #740

___

### **PR Type**
enhancement, bug_fix

___

### **Description**
- Enhanced the 'Create Offer' button in the UI to be disabled if the resource amount to give or get is zero, preventing invalid trade offers.
- Updated the mint resource script to handle command-line arguments and different modes, improving usability and flexibility.
- Updated class hashes in the development manifest files for consistency and correct references.
- Added checks in the trade systems contract to prevent trade orders with zero resource amounts, addressing potential bugs.
- Introduced tests to ensure that trade orders with zero resource amounts are correctly handled, increasing the robustness of the system.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>FastCreateOffer.tsx</strong><dd><code>Enhance validation for creating trade offers</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

client/src/ui/components/cityview/realm/trade/FastCreateOffer.tsx
<li>Added conditions to disable the 'Create Offer' button if the amount of <br>resources to give or get is zero.<br>

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-eb5cd2decc3fb2c33e5be075658d27616b278d831b68c19fe195941d478e2cce">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mint_resource.sh</strong><dd><code>Update mint resource script with argument parsing and mode handling</code></dd></summary>
<hr>

contracts/scripts/testing/mint_resource.sh
<li>Added command-line argument parsing with usage function.<br> <li> Added environment and contract scripts sourcing based on mode.<br> <li> Modified the command structure for resource minting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-361d926cf5a94e2102b531d9d5e41bd315ecdaf1cc463cdc620834ce1ef3c7ec">+26/-1</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>manifest.json</strong><dd><code>Update class hashes in dev manifest</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

contracts/manifests/dev/manifest.json
- Updated class hashes for multiple contracts.

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-bc6816b30ea64e261f152a070b6e767d29d900fcbacb95c684f1c14a82dff5a8">+28/-325</a></td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>manifest.toml</strong><dd><code>Update class hashes in dev manifest TOML</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

contracts/manifests/dev/manifest.toml
- Updated class hashes for multiple contracts.

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-43499a9633d551acabe80100dc29d8995890cd62b12169122658233caea498b3">+28/-45</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>trade_systems.cairo</strong><dd><code>Add zero amount checks in trade systems contract</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

contracts/src/systems/trade/contracts/trade_systems.cairo
<li>Added assertions to ensure resource amounts are non-zero when creating <br>trade orders.<br>

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-e84e5e5d21898c44ad2cd3739a38f25dc14b6b8f1b627db54326aabdb5bcc22d">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>create_order.cairo</strong><dd><code>Add tests for zero resource amount validation in trade orders</code></dd></summary>
<hr>

contracts/src/systems/trade/tests/trade_systems_tests/create_order.cairo
<li>Added tests to verify that creating orders with zero resource amounts <br>fails as expected.<br>

</details>
    

  </td>
  <td><a href="https://github.com/BibliothecaDAO/eternum/pull/756/files#diff-326184473e6d0d5ee7aa64d4ce6f42d666eab8662bed1a5ed4a8749234f85053">+34/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: contracts/src/systems/trade/contracts/trade_systems.cairo [hunk 1]
```diff
@@ -90,6 +90,7 @@ mod trade_systems {
                     Option::Some((
                         resource_type, resource_amount
                     )) => {
+                        assert(*resource_amount != 0, 'maker resource amount is 0');
```
Comment (ponderingdemocritus) [inline contracts/src/systems/trade/contracts/trade_systems.cairo:93]: not sure we need the contract changes in this PR. Since it will fail anyway from the balance being 0, so this check isn't needed.

Comment (aymericdelab) [reply contracts/src/systems/trade/contracts/trade_systems.cairo:93]: i think the issue might be that when you call create_order in the UI with 0 resources it's gonna input an empty list, and that will still create the order nothing inside

Comment (edisontim) [reply contracts/src/systems/trade/contracts/trade_systems.cairo:93]: If balance is `> 0` but `resource_amount == 0` the transaction will pass and an order is created at the moment. So we get weird orders like these haha:
image

> @edisontim can you check the calldata when resource or lords is 0 ?  

Indeed it is

Comment (aymericdelab) [reply contracts/src/systems/trade/contracts/trade_systems.cairo:93]: @ponderingdemocritus @edisontim seems to me that assert is needed then, lgtm
